From 09cdc1cbb887ac99e0e747f176e99fdad30c7c40 Mon Sep 17 00:00:00 2001
From: Eduardo Mendi <eduardox.mendi@intel.com>
Date: Thu, 2 May 2013 11:58:40 +0300
Subject: [PORT FROM MAIN] Telephony: In Double call, ringtone must keep in earpiece

BZ: 80477

[DO NOT UPSTREAM]

The ringtone can be heard in IHF when receive and answer a second voice call.
The root cause is that the function setAudioMode is called upon accepting
a second call while a first call is active. In this case, the audio mode is
MODE_IN_CALL and should remain MODE_IN_CALL. However, in function answerCall
the setAudioMode is called before that the phone state of the waitting phone
is changed from WAITING to ACTIVE. Then the getState function state returns
state RINGING which leads to wrong audio mode configuration.

Change-Id: I4b776d59d0a923eeb9d98a9e0bb3d6d2626889ec
Orig-Change-Id: Ia769f9689ca98cafdd6a1b97b3128496ab0435b2
Signed-off-by: Eduardo Mendi <eduardox.mendi@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/internal/telephony/CallManager.java    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/CallManager.java b/src/java/com/android/internal/telephony/CallManager.java
index b818ffa..9c33a7e 100644
--- a/src/java/com/android/internal/telephony/CallManager.java
+++ b/src/java/com/android/internal/telephony/CallManager.java
@@ -381,7 +381,7 @@ public final class CallManager {
         switch (getState()) {
             case RINGING:
                 int curAudioMode = audioManager.getMode();
-                if (curAudioMode != AudioManager.MODE_RINGTONE) {
+                if ((!hasActiveFgCall()) && (curAudioMode != AudioManager.MODE_RINGTONE)) {
                     // only request audio focus if the ringtone is going to be heard
                     if (audioManager.getStreamVolume(AudioManager.STREAM_RING) > 0) {
                         if (VDBG) Rlog.d(LOG_TAG, "requestAudioFocus on STREAM_RING");
-- 
1.7.4.1

